public/app/features/transformers/calculateHeatmap/utils.test.ts

```diff
@@ -0,0 +1,21 @@
+import { convertDurationToMilliseconds } from './utils';
+
+describe('Heatmap utils', () => {
+  const cases: Array<[string, number | undefined]> = [
+    ['1', 1],
+    ['6', 6],
+    ['2.3', 2],
+    ['1ms', 1],
+    ['5MS', 5],
+    ['1s', 1000],
+    ['1.5s', undefined],
+    ['1.2345s', undefined],
+    ['one', undefined],
+    ['20sec', undefined],
+    ['', undefined],
```

Comment:
I'm not sure how we would feel about fractional values in this. When you enter a number (not duration) with a decimal, it uses `Number.parseInt` which will round for you, so 2.3 will eval to 2. When you put in a duration with a decimal, it will always just return undefined based on the core duration parse logic. I don't think this is a huge deal but it's something to note. Let me know if we should be consistent and not allow decimals ever.